x/posts/keeper/grpc_query.go

```diff
@@ -15,8 +16,8 @@ import (
 
 var _ types.QueryServer = &Keeper{}
 
-// Posts implements the QueryPosts gRPC method
-func (k Keeper) Posts(ctx context.Context, request *types.QueryPostsRequest) (*types.QueryPostsResponse, error) {
+// SubspacePosts implements the QuerySubspacePosts gRPC method
+func (k Keeper) SubspacePosts(ctx context.Context, request *types.QuerySubspacePostsRequest) (*types.QuerySubspacePostsResponse, error) {
```

Comment:
Why does `Posts` split to be `SubspacePosts` and `SectionPosts`? 
Their logic are only different in iteration, could be merged into `Posts` from my side.

Reply:
Initially I implemented it using a single query (`Posts`) as you suggest as well. This however had a major drawback: since the section id `0` is a valid id (and refers to the default section that all subspaces will have), it would have required us to use a `int64` value within the query to allow specifying `section_id = -1` to query the posts on all the subspace no matter what section they are in. Even though it would have worked, I preferred splitting the two queries to make it easier for developers to understand what they do